osc-ui/src/main/java/org/osc/core/broker/view/securityinterface/SecurityGroupInterfaceSubView.java

```diff
@@ -157,17 +160,27 @@ public Object generateCell(CustomTable source, Object itemId, Object columnId) {
             }
         });
 
-        this.table.addGeneratedColumn("policyName", new ColumnGenerator() {
-            @Override
-            public Object generateCell(CustomTable source, Object itemId, Object columnId) {
-                SecurityGroupInterfaceDto dto = SecurityGroupInterfaceSubView.this.tableContainer.getItem(itemId)
-                        .getBean();
-                if (dto.getPolicyName() == null) {
-                    return VmidcMessages.getString(VmidcMessages_.TEXT_NOTAPPLICABLE);
-                }
-                return dto.getPolicyName();
-            }
-        });
+		this.table.addGeneratedColumn("policyName", new ColumnGenerator() {
+			@Override
+			public Object generateCell(CustomTable source, Object itemId, Object columnId) {
+				SecurityGroupInterfaceDto dto = SecurityGroupInterfaceSubView.this.tableContainer.getItem(itemId)
+						.getBean();
+				Set<String> policyNameSet = new HashSet<>();
+				if (dto.getPolicies().size() >= 1) {
+					for (PolicyDto policyDto : dto.getPolicies()) {
```

Comment:
you can use stream to collect policy names. Also this logic seems complicated, 2 places where we are returning TEXT_NOTAPPLICABLE see if can be simplified